ui-input-state/src/pointer_state.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2025 the UI Events Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![allow(missing_docs)]
+
+use alloc::vec::Vec;
+use ui_events::pointer::{PointerButton, PointerButtons, PointerEvent};
+
+/// A stateful view of the primary pointer.
+#[derive(Clone, Debug, Default)]
+pub struct PointerState {
+    events: Vec<PointerEvent>,
+    pub down: PointerButtons,
+}
```

Comment:
I think this could be named differently from `pointer::PointerState`, perhaps something that makes it clear that it's only for the primary pointer. `PrimaryPointer` may be descriptive enough.